### PR TITLE
LIVE-1575: add Headline story

### PR DESCRIPTION
**Pull request by joecowton1**
## Why are you doing this?

Adds Storybook stories for all formats in `Headline` component

## Changes

- add stories
- refactor `Headline` component to use more descriptive function names and put styles into sections

##

To test run storybook and check `editions/Headline` stories

### CODE REVIEW

File: src/components/editions/header.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { from } from '@guardian/src-foundations/mq';
 import { Design, Display } from '@guardian/types';
 import Byline from 'components/editions/byline';
 import HeaderMedia from 'components/editions/headerMedia';
-import Headline from 'components/editions/headline';
```
Comment (abeddow91) [inline src/components/editions/header.tsx:10]: What are your thoughts on having an index.js file in each sub folder that exports the component file? This way we won't need to update the import paths and it keeps the paths cleaner to read.

Comment (joecowton1) [reply src/components/editions/header.tsx:10]: Yeah I'm into this I think. @faresite if you're ok with the idea we can go with that?

Comment (faresite) [reply src/components/editions/header.tsx:10]: makes sense 👍

Comment (joecowton1) [reply src/components/editions/header.tsx]: Updated 👍

File: src/components/editions/headline/headline.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			theme: selectPillar(Pillar.News),
+		}}
+	/>
+);
+
+const Review = (): ReactElement => (
+	<Headline
+		item={{
+			...review,
+			...isImmersive(),
```
Comment (faresite) [inline src/components/editions/headline/headline.stories.tsx:64]: Don't think it's related to this PR but can a Review be immersive? Seems like the star ratings are broken in immersive.

Comment (joecowton1) [reply src/components/editions/headline/headline.stories.tsx:64]: Thats a good question - I think not. I'll remove it to avoid any confusion.

**Review by abeddow91 — APPROVED** [review]

**Pull request merged**